src/DIRAC/Resources/IdProvider/OAuth2IdProvider.py

```diff
@@ -304,128 +433,55 @@ def fetchToken(self, **kwargs):
         self.token["provider"] = self.name
         return S_OK(OAuth2Token(dict(self.token)))
 
-    def revokeToken(self, token=None, token_type_hint="refresh_token"):
-        """Revoke token
-
-        :param str token: token
-        :param str token_type_hint: token type
-
-        :return: S_OK()/S_ERROR()
-        """
-        if not token:
-            token = self.token.get(token_type_hint)
-        try:
-            self.revoke_token(self.get_metadata("revocation_endpoint"), token=token, token_type_hint=token_type_hint)
-        except Exception as e:
-            self.log.exception(e)
-            return S_ERROR(repr(e))
-        return S_OK()
-
-    def exchangeToken(self, group=None, scope=None):
+    def exchangeToken(self, accessToken, group=None, scope=None):
         """Get new tokens for group scope
 
+        :param str accessToken: access token
         :param str group: requested group
         :param list scope: requested scope
 
         :return: dict -- token
         """
+        if not accessToken:
+            return S_ERROR("Access token is empty")
+
         scope = scope or scope_to_list(self.scope)
         if group:
             if not (groupScopes := self.getGroupScopes(group)):
                 return S_ERROR(f"No scope found for {group}")
             scope = list(set(scope + groupScopes))
         scope = list_to_scope(scope)
+
         try:
             token = self.exchange_token(
                 self.get_metadata("token_endpoint"),
-                subject_token=self.token["access_token"],
+                subject_token=accessToken,
                 subject_token_type="urn:ietf:params:oauth:token-type:access_token",
                 scope=scope,
             )
-            if not token:
-                return S_ERROR(f"Cannot exchange token with {scope} scope.")
             return S_OK(OAuth2Token(dict(token)))
-        except Exception as e:
-            self.log.exception(e)
-            return S_ERROR(f"Cannot exchange token with {scope} scope: {repr(e)}")
+        except OAuthError as e:
```

Comment:
Similar comment as above, also for some other places

Reply:
`OAuthError` is the expected exception that we can get from `authlib` here.
But I don't handle the `requests` exceptions that can also occur while interacting with the IdProvider (like I do in `_handleRequest()`.
I added back a general `except Exception` for the moment.

Not really related to this but:
I wonder whether we really need `OAuth2IdProvider` to inherit from `authlib.OAuth2Session`.
I have the feeling that we could just use the `requests` library to interact with the IdProvider (the code I tested does not absolutely require `authlib`, but I haven't deeply checked the code related to the `device_code` and `authorization_code` flows yet).